.github/actions/python_cache/action.yml

```diff
@@ -0,0 +1,53 @@
+name: Python Cache
+description: Caches a full installation of Haystack dependencies
```

Comment:
Can't we use the caching system from `setup-python` directly? https://github.com/actions/setup-python#caching-packages-dependencies

Reply:
Wow sure! I didn't know about this. Soon to be implemented!